### ISSUE TITLE
Preserve extension when uploading files

### ISSUE THREAD

**Issue by sedubois** [graphcool/prisma#180]
My first step to migrate has been to upload many files to Graphcool. However, I can't reference them from my old codebase, because the file extension isn't exposed in the URL, and [a library](http://foundation.zurb.com/sites/docs/v/5.5.3/components/interchange.html) in this old code only understands images with file extensions. This is a [limitation which won't get fixed](https://github.com/zurb/foundation-sites/issues/5512). I might have similar issues with other content like audio and videos.

Even if I only need this in the interim while migrating, it would be great if Graphcool could preserve file extensions, because this procedure will likely take weeks or months. At the moment it therefore appears that I'm blocked.

(This might be a particular example of a more generic need to expose (a subset of) the filename in the URL (or some other metadata such as a "title" or "description"). [E.g Medium](https://blog.graph.cool/announcing-integrations-team-collaboration-and-pricing-updates-7514d5c5686f) exposes URLs which are impossible to guess, but also legible.)

Thanks!

**Comment by sedubois** [comment]
I thought that I got around this issue, but found several places in my legacy codebase which reference the file extension (this time, just found issue in iOS swift code). So I need to switch back my migration towards CloudFront which preserves the file extensions (and which also preserves signed URLs: #182).

**Comment by sedubois** [comment]
OK, I got around this specific issue by adding `?hint=.mp4` to the file 😄

**Comment by kbrandwijk** [comment]
You could also create a small wrapper on webtask.io with the filename in the url I guess.

**Comment by sedubois** [comment]
Thanks @kbrandwijk for the suggestion. I found yet another problematic place (Android client this time which was really expecting an .mp4 extension in the path itself, no good in query params), and after going back and forth with other solutions I figured out I could just use a CloudFlare URL forwarding rule to redirect `my.sub.domain.com/*.mp4?hint=other-string-which-was-also-expected` onto `https://files.graph.cool/$1`... (where $1 matches the *.) Maybe I could have done something similar with webtask.io, but I like CloudFlare and was already using it. 

In case it helps someone else 😄 Considering how much time it took me to figure this out (I'm no expert, which is why I'm interested in Graphcool in the first place), it would really help if Graphcool allowed enriching the URL with a custom string (incl. the ending) and I guess it wouldn't be that complicated.
